pkg/controller/elasticsearch/elasticsearch_controller.go

```diff
@@ -295,6 +310,30 @@ func (r *ReconcileElasticsearch) updateStatus(
 	return common.UpdateStatus(r.Client, cluster)
 }
 
+func (r *ReconcileElasticsearch) annotateResource(
```

Comment:
nit: Maybe add a comment to explain the purpose and the scope of the annotation set by this function.